RyuSocks/Packets/UaPPacket.cs

```diff
@@ -0,0 +1,21 @@
+namespace RyuSocks.Packets
+{
+    public struct UaPPacket(string username, string password) : IAuthenticationPacket
+    {
+        public byte Version { get; set; }
+        public int UsernameLength = username.Length;
+        public string Username = username;
+        public int PasswordLength = password.Length;
+        public string Password = password;
+        public void FromArray(byte[] array)
+        {
+            Version = array[0];
+            UsernameLength = array[1];
+            Username = array[2..UsernameLength].ToString();
```

Comment:
This won't work, as this will only create a string representation of the bytes.
You should probably use `Encoding.ASCII.GetString()` for this (see: `EndpointPacket`).